GVFS/GVFS.Common/InstallerPreRunChecker.cs

```diff
@@ -198,6 +198,7 @@ private bool IsGVFSUpgradeAllowed(out string consoleError)
                     Environment.NewLine,
                     $"{GVFSConstants.UpgradeVerbMessages.GVFSUpgrade} is only supported after the \"Windows Projected File System\" optional feature has been enabled by a manual installation of VFS for Git, and only on versions of Windows that support this feature.",
                     "Check your team's documentation for how to upgrade.");
+                this.tracer.RelatedError($"{nameof(this.IsGVFSUpgradeAllowed)}: Upgrade is not installable. {consoleError}");

```

Comment:
According to my understanding, these should probably not be errors either. Do we want this event reported via telemetry?

//cc @wilbaker

Reply:
@jamill I agree this event shouldn't be an error, and I'm not even sure it should be a warning.  For users running pre-RS4 Windows (or Server 2016) this message is expected and not indicative of any problems.

(I think the telemetry question is a separate one, @jrbriggs do you think this is something we'll want to track?)

Reply:
Updated behavior -
* Unattended mode - Log error locally. No reporting to server.
* Non-elevated upgrade - Log error locally. No reporting to server.
* PrjFlt not inboxed & enabled - Report to server.
* Service not running - Log error locally. No reporting to server.
* Installer blocked by running processes - Log error locally.

Reply:
> PrjFlt not inboxed & enabled - Report to server

@alameenshah apologies I missed in our offline discussion that we'd be logging this as an error even in the scenarios where it's expected.

What's the reason for logging this as an error on pre-RS4 machines?

Reply:
when we spoke we decided to keep the same logging level as the same what we'd get on mount.  Is it error there @alameenshah ?

Reply:
In mount this is treated as error.

GVFSMountProcess.cs line 27
```
if (!EnableAndAttachProjFSHandler.TryEnablePrjFlt(this.tracer, out error))
{
    this.tracer.RelatedError($"{nameof(this.Mount)}: Could not enable PrjFlt: {error}");
}
```

On pre-RS4 PCs, since this is the default PrjFlt state, I'm turning this into a warning, with live-site reporting enabled.

Reply:
> In mount this is treated as error.

@alameenshah FYI the code you referenced (`GVFSMountProcess`) is run by GVFS.Service when auto-mounting, it is not part of `gvfs mount`.

That code (in `GVFSMountProcess`) is _not_ expected to fail on pre-RS4 machines.

Reply:
You might be looking for the mount code here:

https://github.com/Microsoft/VFSForGit/blob/4baa37df6bde2c9a9e1917fc7ce5debd653777c0/GVFS/GVFS/CommandLine/MountVerb.cs#L120

Please note that `gvfs mount` does not care whether or not ProjFS is inbox, it only cares that it can be enabled and attached to the volume.

`gvfs upgrade` might be the only verb that behaves differently (from the user's perspective) based on whether or not they're pre-RS4).